content/posts/2019/08/libplanet-0.5.eng.md

```diff
@@ -0,0 +1,106 @@
+
+However, if you have a reliable node, it is quite possible to effectively reduce computing time by using precalculated results from that node.
+
+So, from Version 0.5, by handing over trusted nodes as parameters to [‘Swarm<T>.PreloadAsync()’], we’re now able to receive, store, and use the most recent state values already computed from those trusted nodes.
+
+If there aren’t any trusted nodes or the process fails for another reason, IBD computes in the same process as previous versions.
+
+[IBD]: https://bitcoin.org/en/glossary/initial-block-download
+[`Swarm<T>.PreloadAsync()`]: https://docs.libplanet.io/0.5.0/api/Libplanet.Net.Swarm-1.html#Libplanet_Net_Swarm_1_PreloadAsync_System_IProgress_Libplanet_Net_PreloadState__System_Collections_Immutable_IImmutableSet_Libplanet_Address__System_Threading_CancellationToken_
+
+
+`IRandom.NextDouble()` Removed
+------------------------
+
+Quoting from the [System.Double section of .NET Standard Official Document][official-docs], it states: 
+
+>In addition, the result of arithmetic and assignment operations with ‘Double’ values may differ slightly by platform because of the loss of precision of the ‘Double’ type. For example, the result of assigning a literal ‘Double’ value may differ in the 32-bit and 64-bit versions of the .NET Framework. 
+
+As stated above, arithmetic operations and substitution of Double can cause indeterminant results. Because of this, we’ve decided not to provide [‘IRandom.NextDouble()’] from this version onwards.
+
+Please refer to [this article][floating-point-determinism] for further information.
+
+[`IRandom.NextDouble()`]: https://github.com/planetarium/libplanet/pull/419
+[official-docs]: https://docs.microsoft.com/en-us/dotnet/api/system.double?view=netstandard-2.0#remarks
+[floating-point-determinism]: https://randomascii.wordpress.com/2013/07/16/floating-point-determinism/
+
+
+Block Action
+-------
+
+In order to keep chains secure, we need miners to mine the blocks, and we need to reward miners to recruit and keep them.
+
+Previously for miners to be rewarded, they had to add a transaction with reward actions to the block each time. But from this version, the new [‘BlockPolicy<T>.BlockAction’] property enables miners to implement a code that rewards them for each block action. 
+
+[`BlockPolicy<T>.BlockAction`]: https://docs.libplanet.io/0.5.0/api/Libplanet.Blockchain.Policies.BlockPolicy-1.html#Libplanet_Blockchain_Policies_BlockPolicy_1_BlockAction
+
+
+`FileStore` Removed
+---------------
+
+Libplanet had been providing an interface called [‘IStore’], [‘FileStore’] (a file-based implementation of IStore), as well as [‘LiteDBStore’] (implementation based on [LiteDB]) since Version [0.4] to simplify the storage layer. Despite the benefits of its simplistic implementation, however, ‘FileStore’ also had the following limitations:
```

Comment:
```suggestion
Libplanet had been providing an interface called [`IStore`], [`FileStore`][] (a file-based implementation of `IStore`), as well as [`LiteDBStore`][] (implementation based on [LiteDB]) since Version [0.4] to simplify the storage layer. Despite the benefits of its simplistic implementation, however, `FileStore` also had the following limitations:
```